### PR TITLE
[MNT] Refactor legacy test config

### PR DESCRIPTION
**Pull request by lmmentel**
#### What does this implement/fix? Explain your changes.

- replace legacy `check_` with `test_` in test configuration
- exclude `test_fit_does_not_overwrite_hyper_params` for `FeatureUnion`
- small refactoring of the tests

#### Does your contribution introduce a new dependency? If yes, which one?

No.

#### What should a reviewer concentrate their feedback on?

Check if all the test are running.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

File: sktime/tests/test_all_estimators.py [hunk 1]
```diff
@@ -168,23 +165,26 @@ def test_estimator_tags(estimator_class):
...
 
     # Usually estimators inherit only from one BaseEstimator type, but in some cases
     # they may be predictor and transformer at the same time (e.g. pipelines)
-    n_base_types = sum(
-        [issubclass(Estimator, cls) for cls in VALID_ESTIMATOR_BASE_TYPES]
+    assert (
+        2
+        >= (
+            sum(
+                [issubclass(estimator_class, cls) for cls in VALID_ESTIMATOR_BASE_TYPES]
```
Comment (fkiraly) [inline sktime/tests/test_all_estimators.py:178]: I think this change makes the assertion more obscure to understand. Bracket overload...
Would you feel strongly about reverting it to its original state?

**Review by fkiraly — APPROVED** [review]
Thanks a lot!

Agree with most of the "side changes", they improve style and readability ... except the one where you introduce 100 brackets. Do you maybe want to revert this? Not feeling strongly though.

File: sktime/tests/test_all_estimators.py (hunk 1 above)
Comment (lmmentel) [reply sktime/tests/test_all_estimators.py:178]: This was autorefactored by an extension I use. I can revert back if it bother you.

**Review by fkiraly — APPROVED** [review]
Thanks!.
Unix test failures do not seem content/change related.

**Pull request merged**